_source/_posts/2022-10-01-spring-vault.md

```diff
@@ -0,0 +1,639 @@
+  bootVersion==2.7.4 \
+  dependencies==cloud-config-server \
+  groupId==com.okta.developer \
+  artifactId==vault-config-server  \
+  name=="Spring Boot Configuration Server" \
+  description=="Demo project of a Spring Boot application with Vault protected secrets" \
+  packageName==com.okta.developer.vault > vault-config-server.zip
+```
+
+Unzip the downloaded file. Rename `src/main/resource/application.properties` to `src/main/resource/application.yml`, edit the file to specify the port, add a `native` profile, and specify config search locations:
+
+```yml
+server:
+  port: 8888
+
+spring:
+  profiles:
+    active: native
+```
+
+Edit `src/main/java/com/okta/developer/vault/SpringBootConfigurationServerApplication.java` and add a `@EnableConfigServer` annotation:
+
+```java
+package com.okta.developer.vault;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.config.server.EnableConfigServer;
+
+@EnableConfigServer
+@SpringBootApplication
+public class SpringBootConfigurationServerApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(SpringBootConfigurationServerApplication.class, args);
+    }
+
+}
+```
+
+Start the server, as you are going to encrypt your Okta secrets using the `/encrypt` endpoint. For this example, you are using a symmetric (shared) encryption key, passed through the environment variable ENCRYPT_KEY. Before running the command below, you should replace `{encryptKey}` with a random string of characters. You can use JShell to generate a UUID you can use for your encrypt key.
+
+```shell
+jshell
+
+UUID.randomUUID()
+```
+
+```shell
+ENCRYPT_KEY={encryptKey} ./mvnw spring-boot:run
+```
+
+Then, in another terminal, encrypt your client ID and secret.
+
+```shell
+http :8888/encrypt --raw {yourOktaClientId}
+http :8888/encrypt --raw {yourOktaClientSecret}
+```
+
+In the `vault-config-server` project folder, create a `src/main/resources/config/vault-demo-app-dev.yml` file to store the secrets for the `dev` profile, with the following contents:
+
+```yml
+okta:
+  oauth2:
+    issuer: {yourIssuerURI}
+    clientId: '{cipher}encryptedClientId'
+    clientSecret: '{cipher}encryptedClientSecret'
+```
+
+The `client-id` and `client-secret` encrypted values must be prefixed with `{cipher}`. Restart the config server.
+
+To consume the config server properties, the client application must set the server address in the application configuration. In the `vault-demo-app` project, rename `application.properties` to `application.yml` and configure it as follows:
+
+```yaml
+spring:
+  cloud:
+    config:
+      uri: http://localhost:8888
+  config:
+    import: "configserver:"
+  application:
+    name: vault-demo-app
+  profiles:
+    active: dev
+```
+
+Start `vault-demo-app` without passing the environment variables:
+
+```shell
+./mvnw clean
+./mvnw spring-boot:run
```

Comment:
You can combine these into one command, or leave them as is. I did confirm that clean is necessary.

Reply:
I combined